### ISSUE TITLE
Docker environment throws 500 because it cant find `notebook` module

### ISSUE THREAD

**Issue by zbyte64** [jupyterhub/jupyterhub#879]
**How to reproduce the issue**
Run the jupyterhub docker image:

`docker run -P -d --name jupyterhub -v $PWD:/srv/jupyterhub -t jupyterhub/jupyterhub`

Create a user as stated in the readme:

```
docker exec -it jupyterhub bash
useradd student
passwd student
``` 

Go to the local jupyterhub in the browser and login.

**What you expected to happen**

Upon successful login we should see a jupyter notebook.

**What actually happens**

We get a 500 error:

```
500 : Internal Server Error
Spawner failed to start [status=1]
```

The logs show:

```
[I 2016-11-22 00:58:13.974 JupyterHub spawner:633] Spawning jupyterhub-singleuser '--user="student"' '--cookie-name="jupyter-hub-token-student"' '--base-url="/user/student"' '--hub-host=""' '--hub-prefix="/hub/"' '--hub-api-url="http://127.0.0.1:8081/hub/api"' '--ip="127.0.0.1"' --port=51639
Couldn't set CWD to /home/student ([Errno 2] No such file or directory: '/home/student')
Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/jupyterhub/singleuser.py", line 15, in <module>
    import notebook
ImportError: No module named 'notebook'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/jupyterhub-singleuser", line 3, in <module>
    from jupyterhub.singleuser import main
  File "/opt/conda/lib/python3.5/site-packages/jupyterhub/singleuser.py", line 17, in <module>
    raise ImportError("JupyterHub single-user server requires notebook >= 4.0")
ImportError: JupyterHub single-user server requires notebook >= 4.0
```

**Share what version of JupyterHub you are using**

`0.7.0.b1`

Running `jupyter troubleshoot` in the docker container results in `command not found`

**Comment by minrk** [comment]
The `jupyterhub/jupyterhub` image only has the dependencies to run JupyterHub, not single-user servers. If you want to run single-user servers in the same container as the Hub, you will need to add the notebook package. Traditionally, the jupyterhub container is used in conjunction with a Spawner that puts single-user servers in other containers, such as DockerSpawner.

**Comment by willingc** [comment]
Closing this issue as resolved after PR #915.
